### PR TITLE
add actor provider

### PR DESCRIPTION
**Pull request by BrandonArp**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -231,6 +231,12 @@
       <version>${guice.version}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>javax.inject</groupId>
+      <artifactId>javax.inject</artifactId>
+      <version>1</version>
```
Comment (vjkoskela) [inline pom.xml]: Property for version

File: src/main/java/com/arpnetworking/commons/akka/ActorProvider.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+/**
+ * A dependency injection {@link Provider} to assist in the creation of Akka actors.
+ *
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public class ActorProvider implements Provider<ActorRef> {
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/commons/akka/ActorProvider.java]: Final?

File: src/main/java/com/arpnetworking/commons/akka/ActorProvider.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        _name = name;
+        _clazz = clazz;
+        _actorRefFactory = actorRefFactory;
+    }
+
+    private final String _name;
+    private final Class<? extends Actor> _clazz;
+    private final ActorRefFactory _actorRefFactory;
+
+    /** package private for test */ Injector getInjector() {
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/commons/akka/ActorProvider.java]: Should be above members

File: src/test/java/com/arpnetworking/commons/akka/ActorProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * Tests for the ActorProvider class.
+ *
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public class ActorProviderTest {
```
Comment (vjkoskela) [inline src/test/java/com/arpnetworking/commons/akka/ActorProviderTest.java]: Final

File: pom.xml (hunk 1 above)
Comment (BrandonArp) [reply pom.xml]: Always going to be 1 because its built-in to the jdk

File: src/main/java/com/arpnetworking/commons/akka/ActorProvider.java (hunk 2 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/commons/akka/ActorProvider.java]: sure

File: src/main/java/com/arpnetworking/commons/akka/ActorProvider.java (hunk 3 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/commons/akka/ActorProvider.java]: fixed

File: src/test/java/com/arpnetworking/commons/akka/ActorProviderTest.java (hunk 4 above)
Comment (BrandonArp) [reply src/test/java/com/arpnetworking/commons/akka/ActorProviderTest.java]: sure.... but I think you're taking this final class thing a bit too far

**Pull request merged**